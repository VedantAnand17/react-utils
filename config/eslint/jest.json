{
  "env": {
    "browser": true,
    "jest": true
  },
  "plugins": [
    "jest"
  ],
  "rules": {
    "@typescript-eslint/no-implied-eval": 0,
    "global-require": 0,
    "import/no-dynamic-require": 0,
    "import/no-extraneous-dependencies": 0,
    "no-console": 0,
    "no-new-func": 0,
    "react/require-default-props": 0
  }
}