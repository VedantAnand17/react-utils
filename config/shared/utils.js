/**
 * Misc utilities reusable in different JS-format config files.
 */
/* eslint-disable import/no-extraneous-dependencies */

const fs = require('fs');
const path = require('path');
const cssesc = require('cssesc');

const { interpolateName } = require('loader-utils');

/**
 * Normalizes file path to OS-independent format (adopted from css-loader).
 * @ignore
 * @param {string} file
 * @return {string}
 */
function normalizePath(file) {
  return path.sep === '\\' ? file.replace(/\\/g, '/') : file;
}

// eslint-disable-next-line no-control-regex
const filenameReservedRegex = /[<>:"/\\|?*]/g;
// eslint-disable-next-line no-control-regex
const reControlChars = /[\u0000-\u001f\u0080-\u009f]/g;

function escapeLocalident(localident) {
  return cssesc(
    localident
      // For `[hash]` placeholder
      .replace(/^((-?[0-9])|--)/, '_$1')
      .replace(filenameReservedRegex, '-')
      .replace(reControlChars, '-')
      .replace(/\./g, '-'),
    { isIdentifier: true },
  );
}

/**
 * Returns the name of package containing the folder; i.e. it recursively looks
 * up from the folder for the closest package.json file, and returns the name in
 * that file. It also caches the results from previously fisited folders.
 * @ignore
 * @param {string} folder
 * @return {string}
 */
function getPackageInfo(folder) {
  let res = getPackageInfo.cache[folder];
  if (!res) {
    const p = path.resolve(folder, 'package.json');
    /* eslint-disable global-require, import/no-dynamic-require */
    res = fs.existsSync(p) ? {
      name: require(p).name,
      root: folder,
    } : getPackageInfo(path.resolve(folder, '..'));
    /* eslint-enable global-require, import/no-dynamic-require */
    getPackageInfo.cache[folder] = res;
  }
  return res;
}
getPackageInfo.cache = {};

function getLocalIdent(
  { resourcePath },
  localIdentName,
  localName,
  options = {},
) {
  const packageInfo = getPackageInfo(path.dirname(resourcePath));
  const request = normalizePath(path.relative(packageInfo.root, resourcePath));
  return interpolateName({
    resourcePath,
  }, localIdentName, {
    ...options,
    content: `${packageInfo.name + request}\x00${localName}`,
    context: packageInfo.root,
  }).replace(/\[package\]/gi, packageInfo.name)
    .replace(/\[local\]/gi, localName)
    .replace(/@/g, '-');
}

function generateScopedNameDev(localName, assetPath) {
  // Mind: "css-loader" additionally escapes identifiers returned by
  // getLocalIdent(); the babel plugins - don't, thus we need to escape
  // ourselves.
  return escapeLocalident(
    getLocalIdent(
      { resourcePath: assetPath },
      '[package]___[path][name]___[local]___[hash:base64:6]',
      localName,
      {},
    ),
  );
}

function generateScopedNameProd(localName, assetPath) {
  return escapeLocalident(
    getLocalIdent(
      { resourcePath: assetPath },
      '[hash:base64:6]',
      localName,
      {},
    ),
  );
}

module.exports = {
  generateScopedNameDev,
  generateScopedNameProd,
  getLocalIdent,
};
