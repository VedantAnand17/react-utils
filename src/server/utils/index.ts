// Aggregates utility modules together.

export * as errors from './errors';
