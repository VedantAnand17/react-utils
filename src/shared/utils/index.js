import path from 'path';

import config from './config';
import * as isomorphy from './isomorphy';
import * as redux from './redux';
import * as webpack from './webpack';

const juUrl = module.webpackPolyfill ? './shared/utils/jest'
  : path.resolve(__dirname, './jest');

const JU = isomorphy.isServerSide() ? webpack.requireWeak(juUrl) : null;

export {
  config,
  isomorphy,
  JU,
  redux,
  webpack,
};
