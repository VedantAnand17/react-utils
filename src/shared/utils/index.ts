import themedImpl, {
  type Theme,
  COMPOSE,
  PRIORITY,
  ThemeProvider,
} from '@dr.pogodin/react-themes';

import config from './config';
import * as isomorphy from './isomorphy';
import time from './time';
import * as webpack from './webpack';

import type * as JuT from './jest';

export {
  Barrier,
  Emitter,
  Semaphore,
  withRetries,
} from '@dr.pogodin/js-utils';

export { getSsrContext } from './globalState';
export { default as splitComponent } from './splitComponent';

type ThemedT = typeof themedImpl & {
  COMPOSE: typeof COMPOSE;
  PRIORITY: typeof PRIORITY;
};

const themed: ThemedT = themedImpl as ThemedT;

themed.COMPOSE = COMPOSE;
themed.PRIORITY = PRIORITY;

// Note: it should be done this way, as in some environments
// "process" might not exist, and process.env.NODE_CONFIG_ENV
// not injected by Webpack.
let NODE_CONFIG_ENV;
try {
  NODE_CONFIG_ENV = process.env.NODE_CONFIG_ENV;
} catch { /* noop */ }

const env = NODE_CONFIG_ENV || process.env.NODE_ENV;

const JU: typeof JuT | null = env !== 'production'
  ? webpack.requireWeak('./jest', __dirname) as typeof JuT | null
  : null;

export {
  type Theme,
  config,
  isomorphy,
  JU,
  themed,
  ThemeProvider,
  time,
  webpack,
};
