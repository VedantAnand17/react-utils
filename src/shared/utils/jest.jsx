/**
 * Utilities for Jest tests.
 */

/* NOTE: At the moment here is no intention to use this module for runtime
 * execution, only for tests; thus we depend on Jest environment and dev
 * dependencies. */

/* TODO: Revise.  */

/* global expect, jest, document */
/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';

import mockdate from 'mockdate';
import PT from 'prop-types';
import { render as newRender, unmountComponentAtNode } from 'react-dom';
import TU from 'react-dom/test-utils';

/* eslint-disable import/no-extraneous-dependencies */
import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
/* eslint-enable import/no-extraneous-dependencies */

/* NEW TESTING METHODS */

export { act } from 'react-dom/test-utils';

const originalProcessVersions = process.versions;

export function mockClientSide() {
  Object.defineProperty(process, 'versions', { value: null });
}

export function unmockClientSide() {
  Object.defineProperty(process, 'versions', {
    value: originalProcessVersions,
    writable: false,
  });
}

/**
 * Generates a mock UUID.
 * @param {Number} seed
 * @return {String}
 */
export function getMockUuid(seed = 0) {
  const x = seed.toString(16).padStart(32, '0');
  return `${x.slice(0, 8)}-${x.slice(8, 12)}-${x.slice(12, 16)}-${x.slice(16, 20)}-${x.slice(20)}`;
}

/**
 * Advances mock timers, and mock date by the specified time.
 * @param {Number} time Time step [ms].
 * @returns {Promise} Wait for this to "jump after" any async code which should
 *  be executed because of the mock time movement.
 */
export async function mockTimer(time) {
  mockdate.set(time + Date.now());
  jest.advanceTimersByTime(time);
}

/**
 * Mounts `Scene` to the DOM, and returns the root scene element.
 * @param {React.ReactNode} scene
 * @return {HTMLElement}
 */
export function mount(scene) {
  const res = document.createElement('div');
  document.body.appendChild(res);
  newRender(scene, res);
  return res;
}

/**
 * Unmounts `scene` from the DOM.
 * @param {HTMLElement} scene
 */
export function unmount(scene) {
  unmountComponentAtNode(scene);
  scene.remove();
}

/* OLD STUFF BELOW THIS MARK */

/**
 * Just an alias for TU.findRenderedDOMComponentWithClass(..).
 * @param {Object} dom
 * @param {String} className
 * @return {Object}
 */
export function findInDomByClass(dom, className) {
  return TU.findRenderedDOMComponentWithClass(dom, className);
}

/**
 * Just an alias for TU.scryRenderedDOMComponentsWithClass(..).
 * @param {Object} dom
 * @param {Stirng} className
 * @return {Array}
 */
export function findInDomManyByClass(dom, className) {
  return TU.scryRenderedDOMComponentsWithClass(dom, className);
}

/**
 * Auxiliary wrapper around ReactJS Test Renderer.
 * @param {Object} component ReactJS component to render.
 * @return {Object} JSON representation of the rendered tree.
 */
export function render(component) {
  return Renderer.create(component).toJSON();
}

/* The Wrapper is necessary for the "renderDom(..)" function, because
 * the "renderIntoDocument(..)" function from "react-dom/test-utils" works
 * only with state components, so we have to wrap our ReactJS components into
 * such Wrapper. */
class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    const { children } = this.props;
    return children;
  }
}

Wrapper.propTypes = {
  children: PT.node.isRequired,
};

/**
 * Renders given component into DOM, using "react-dom/test-utils".
 * @param {Object} component ReactJS component to render.
 * @return {Object} Rendered DOM.
 */
export function renderDom(component) {
  return TU.renderIntoDocument((
    <Wrapper>
      {component}
    </Wrapper>
  ));
}

/**
 * Auxiliary wrapper around ReactJS Shallow Test Renderer.
 * @param {Object} component ReactJS component to render.
 * @return {Object} JSON representation of the shallow component's render tree.
 */
export function shallowRender(component) {
  const renderer = new ShallowRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

/**
 * Renders provided ReactJS component with ReactJS Shallow Test Renderer,
 * creates/tests its snapshot, and returns the render result.
 * @param {Object} component ReactJS component to render.
 * @return {Object} JSON representation of shallow render.
 */
export function shallowSnapshot(component) {
  const res = shallowRender(component);
  expect(res).toMatchSnapshot();
  return res;
}

/**
 * Renders provided ReactJS component with ReactJS Test Renderer,
 * creates/tests the snapshot, and returns the render.
 * @param {Object} component ReactJS component to render.
 * @return {Object} JSON render of the component.
 */
export function snapshot(component) {
  const res = render(component);
  expect(res).toMatchSnapshot();
  return res;
}

export const simulate = TU.Simulate;
