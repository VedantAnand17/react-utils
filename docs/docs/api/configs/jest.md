# Jest
Default configuration for [Jest] testing framework.

:::caution Beware
It does strong assumptions about the host project's code structure: the host
[Jest] configuration in the `/config/jest` folder of host code; the source code
in `/src` folder; build outputs in `/build` folder; _etc._

For a project with different structure it might be easier to compose
the appropriate Jest config from scratch.
:::

To use this default configuration create two files in the `/config/jest` folder
of host project:

```js title="/config/jest/config.js"
const config = require('@dr.pogodin/react-utils/config/jest/default');

// You can tune config as necessary here.

module.exports = config;
```

```js title="/config/jest/setup.js"
import '@dr.pogodin/react-utils/config/jest/setup';
```

After that point to that host config file from the Jest execution command,
_e.g._:
```bash
jest --config config/jest/config.js
```

[Jest]: https://jestjs.io
