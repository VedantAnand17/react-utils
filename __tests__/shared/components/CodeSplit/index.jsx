/**
 * Here are the actual code splitting tests.
 */

import _ from 'lodash';
import mockdate from 'mockdate';
import pretty from 'pretty';

import { act, mockClientSide, unmockClientSide } from 'utils/jest';

let factory;
let Helmet;
let SampleCodeSplit;

jest.mock('node-forge');
jest.mock('uuid');

jest.useFakeTimers();
mockdate.set('2019-11-29Z');

const TEST_HTTP_REQUEST = {
  nonce: 'abcdef-dummy-nonce',
  info: 'I am a dummy HTTP request! No need for a complex mock here!',
};

const TEST_WEBPACK_CONFIG = {
  context: `${__dirname}/test_data`,
  output: {
    path: '/test/path',
    publicPath: '/test/public/path/',
  },
};

/**
 * Reproduces the core of server-side rendering setup.
 * @param {Object} Scene
 * @param {Number} [maxSsrRounds]
 */
async function renderServerSide(Scene, maxSsrRounds) {
  const { IS_SERVER_SIDE } = require('utils/isomorphy');
  expect(IS_SERVER_SIDE).toBe(true);
  const renderer = factory(
    _.cloneDeep(TEST_WEBPACK_CONFIG),
    {
      Application: Scene,
      maxSsrRounds,
      ssrTimeout: Number.MAX_VALUE,
    },
  );
  let render = '';
  await renderer(
    _.cloneDeep(TEST_HTTP_REQUEST),
    {
      locals: {
        webpack: {
          // TODO: Rather than mocking the middleware, use it, to safeguard
          // against any breaking changes in future middleware releases.
          devMiddleware: {
            stats: {
              toJson: () => ({
                namedChunkGroups: {
                  main: {
                    assets: [
                      { name: 'main-1511941200000.css' },
                      { name: 'main-1511941200000.js' },
                    ],
                  },
                  polyfills: {
                    assets: [{ name: 'polyfills-1511941200000.js' }],
                  },
                  'sample-component-1': {
                    assets: [{ name: 'sample-component-1-1511941200000.css' }],
                  },
                  'sample-component-2': {
                    assets: [
                      { name: 'sample-component-2-1511941200000.js' },
                      { name: 'sample-component-2-1511941200000.css' },
                    ],
                  },
                  'sample-component-3': {
                    assets: [
                      { name: 'sample-component-3-1511941200000.js' },
                      { name: 'sample-component-3-1511941200000.css' },
                    ],
                  },
                },
              }),
            },
          },
        },
      },
      send: (x) => { render += x; },
      status: (x) => { render += `HTTP STATUS: ${x}\n`; },
    },
    (error) => expect(error).toMatchSnapshot(),
  );
  return render;
}

afterAll(() => {
  Helmet.canUseDOM = true;
});

/**
 * Resets React environment.
 */
function resetModules(clientSide) {
  if (clientSide) mockClientSide();
  else unmockClientSide();
  jest.resetModules();
  factory = require('server/renderer').default;
  ({ Helmet } = require('react-helmet'));
  Helmet.canUseDOM = false;
  SampleCodeSplit = require(
    'components/CodeSplit/__mocks__/SampleCodeSplit',
  ).default;
}

beforeEach(async () => {
  resetModules();
});

test('Server-side rendering', async () => {
  const { IS_SERVER_SIDE } = require('utils/isomorphy');
  expect(IS_SERVER_SIDE).toBe(true);
  let markup = await renderServerSide(SampleCodeSplit, 1);
  expect(pretty(markup)).toMatchSnapshot();
  markup = renderServerSide(SampleCodeSplit, 3);
  jest.runAllTimers();
  markup = await markup;
  expect(pretty(markup)).toMatchSnapshot();
});

test('Client-side rendering', async () => {
  const { IS_SERVER_SIDE } = require('utils/isomorphy');
  expect(IS_SERVER_SIDE).toBe(true);
  let serverMarkup = renderServerSide(SampleCodeSplit, 3);
  await jest.runAllTimers();
  serverMarkup = await serverMarkup;
  document.write(serverMarkup);
  document.close();
  const ssrHead = pretty(document.head.innerHTML);
  const ssrBody = pretty(document.body.innerHTML);
  expect(ssrHead).toMatchSnapshot();
  expect(ssrBody).toMatchSnapshot();
  window.TRU_KEEP_INJ_SCRIPT = true;
  require('client/init');
  resetModules(true);
  const { IS_CLIENT_SIDE } = require('utils/isomorphy');
  expect(IS_CLIENT_SIDE).toBe(true);
  let Launch = require('client').default;
  act(() => Launch({ getApplication: () => SampleCodeSplit }));
  let head = pretty(document.head.innerHTML);
  let body = pretty(document.body.innerHTML);
  expect(head).toEqual(ssrHead);
  expect(body).toEqual(ssrBody);

  /* This tests that in the real render the data injection script
   * is auto-removed from the document during the injection. */
  resetModules(true);
  document.write(serverMarkup);
  document.close();
  delete window.TRU_KEEP_INJ_SCRIPT;
  require('client/init');
  Launch = require('client').default;
  act(() => Launch({ getApplication: () => SampleCodeSplit }));
  head = pretty(document.head.innerHTML);
  body = pretty(document.body.innerHTML);
  expect(head).toMatchSnapshot();
  expect(body).toMatchSnapshot();
});
