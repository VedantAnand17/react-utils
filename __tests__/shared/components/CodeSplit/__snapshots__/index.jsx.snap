// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client-side rendering 1`] = `
"<title data-react-helmet=\\"true\\"></title>

<meta name=\\"theme-color\\" content=\\"#FFFFFF\\">

<link href=\\"/test/public/path/main-1511941200000.css\\" rel=\\"stylesheet\\">

<meta charset=\\"utf-8\\">
<meta content=\\"width=device-width,initial-scale=1.0\\" name=\\"viewport\\">"
`;

exports[`Client-side rendering 2`] = `
"<div id=\\"react-view\\">
  <h1>testValue</h1>
  <div data-chunk-name=\\"sample-component-1\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
  <div data-chunk-name=\\"sample-component-2\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
  <div data-chunk-name=\\"sample-component-3\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
</div>
<script id=\\"inj\\" type=\\"application/javascript\\" nonce=\\"abcdef-dummy-nonce\\">
  window.INJ = \\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghsfwaFoaus6qLoke9MlgqF39wUyiJOuxKmqjECLiFel3AfVxBlTkA2sj6FNYLq9mBqrVlCGhUyhXTWTPFlSbgmX7mf9UJCV6o+6N49TRM8T/Zt5SRase/W0YFEHYmiKSLZgUQPXsUoxsSTABdZa9BdRV/oUK68Qi2md0HR0J+b+HTZLAIsrRxHKMjBsUfmWUQjhOXxz8uZkjMyqgSptcNXEftmR5kazR/qzkkQcObSKPuDAKU03KPvQYU8IFE4CiNOX2Iw+T3MoeOLcuMcbBkvEUpXOo8tg7/9XQIY2QxEY+VOgOOywhCiFCspUah8KD+PeKXkuJdPx731E90C7TJz6AAvBfqtI2xCmRQrm5eP/Hk+E+WziLt4iuX0A+KQf2GhT7h2B6WWz+OiLd4i7p2hLaS8dJOsWjmKmAPz8mqTlA=\\"
</script>
<script src=\\"/test/public/path/polyfills-1511941200000.js\\" type=\\"application/javascript\\"></script>
<script src=\\"/test/public/path/main-1511941200000.js\\" type=\\"application/javascript\\"></script>"
`;

exports[`Client-side rendering 3`] = `
"<title data-react-helmet=\\"true\\"></title>

<meta name=\\"theme-color\\" content=\\"#FFFFFF\\">

<link href=\\"/test/public/path/main-1511941200000.css\\" rel=\\"stylesheet\\">

<meta charset=\\"utf-8\\">
<meta content=\\"width=device-width,initial-scale=1.0\\" name=\\"viewport\\">"
`;

exports[`Client-side rendering 4`] = `
"<div id=\\"react-view\\">
  <h1>testValue</h1>
  <div data-chunk-name=\\"sample-component-1\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
  <div data-chunk-name=\\"sample-component-2\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
  <div data-chunk-name=\\"sample-component-3\\">
    <div>
      <h1>Sample Component</h1><pre>{
  \\"test\\": {
    \\"key\\": \\"testValue\\"
  },
  \\"dr_pogodin_react_utils___split_components\\": {
    \\"sample-component-1\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    },
    \\"sample-component-2\\": {
      \\"data\\": {},
      \\"numRefs\\": 0,
      \\"operationId\\": \\"\\",
      \\"timestamp\\": 1574985600000
    }
  }
}</pre>
    </div>
  </div>
</div>

<script src=\\"/test/public/path/polyfills-1511941200000.js\\" type=\\"application/javascript\\"></script>
<script src=\\"/test/public/path/main-1511941200000.js\\" type=\\"application/javascript\\"></script>"
`;

exports[`Server-side rendring 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

  <head>

    <title data-react-helmet=\\"true\\"></title>

    <meta name=\\"theme-color\\" content=\\"#FFFFFF\\" />

    <link href=\\"/test/public/path/main-1511941200000.css\\" rel=\\"stylesheet\\" />

    <meta charset=\\"utf-8\\" />
    <meta content=\\"width=device-width,initial-scale=1.0\\" name=\\"viewport\\" />
  </head>

  <body>

    <div id=\\"react-view\\">
      <h1>testValue</h1>
      <div data-chunk-name=\\"sample-component-1\\"></div>
      <div data-chunk-name=\\"sample-component-2\\">
        <div>PLACEHOLDER!</div>
      </div>
      <div data-chunk-name=\\"sample-component-3\\">
        <div>
          <h1>Sample Component</h1><pre>{
  &quot;test&quot;: {
    &quot;key&quot;: &quot;testValue&quot;
  },
  &quot;dr_pogodin_react_utils___split_components&quot;: {
    &quot;sample-component-1&quot;: {
      &quot;data&quot;: null,
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;00000000-0000-0000-0000-000000000001&quot;,
      &quot;timestamp&quot;: 0
    },
    &quot;sample-component-2&quot;: {
      &quot;data&quot;: null,
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;00000000-0000-0000-0000-000000000002&quot;,
      &quot;timestamp&quot;: 0
    }
  }
}</pre>
        </div>
      </div>
    </div>
    <script id=\\"inj\\" type=\\"application/javascript\\" nonce=\\"abcdef-dummy-nonce\\">
      window.INJ = \\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghsfwaFoaus6qLoke9MlgqF39wUyiJOuxKmqjECLiFel3AfVxBlTkA2sj6FNYLq9mBqrVlCGhUyhXTWTPFlSbgmX7mf9UJCV6o+6N49TRM8T/Zt5SRase/W0YFEHYmiKSLZgUQPXsUoxsSTABdZa9BdRV/oUK68Qi2md0HR0J+b+HTZLAIsrRxHKMjBsUfmWUQjhOXxz8uZkjMyqgSptcNXEftmR5kazR/qzkkQcObSKPuDAKU03KPvQYU8IFE4CiNOX2Iw+T3MoeOLcuMcbBkvEUpXOo8tg7/9XQIY2QxEY+VOgOOywhCiFCspUah8KD+PeKXkuJdPx731E90C7TJz6AAvBfqtI2xCmRQrm5eP/Hk+E+WziLt4iuX0A+KQf2GhT7h2B6WWz+OiLd4i7p2hLaS8dJOsWjmKmAPz8mqTlA=\\"
    </script>
    <script src=\\"/test/public/path/polyfills-1511941200000.js\\" type=\\"application/javascript\\"></script>
    <script src=\\"/test/public/path/main-1511941200000.js\\" type=\\"application/javascript\\"></script>

  </body>

</html>"
`;

exports[`Server-side rendring 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

  <head>

    <title data-react-helmet=\\"true\\"></title>

    <meta name=\\"theme-color\\" content=\\"#FFFFFF\\" />

    <link href=\\"/test/public/path/main-1511941200000.css\\" rel=\\"stylesheet\\" />

    <meta charset=\\"utf-8\\" />
    <meta content=\\"width=device-width,initial-scale=1.0\\" name=\\"viewport\\" />
  </head>

  <body>

    <div id=\\"react-view\\">
      <h1>testValue</h1>
      <div data-chunk-name=\\"sample-component-1\\">
        <div>
          <h1>Sample Component</h1><pre>{
  &quot;test&quot;: {
    &quot;key&quot;: &quot;testValue&quot;
  },
  &quot;dr_pogodin_react_utils___split_components&quot;: {
    &quot;sample-component-1&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    },
    &quot;sample-component-2&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    }
  }
}</pre>
        </div>
      </div>
      <div data-chunk-name=\\"sample-component-2\\">
        <div>
          <h1>Sample Component</h1><pre>{
  &quot;test&quot;: {
    &quot;key&quot;: &quot;testValue&quot;
  },
  &quot;dr_pogodin_react_utils___split_components&quot;: {
    &quot;sample-component-1&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    },
    &quot;sample-component-2&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    }
  }
}</pre>
        </div>
      </div>
      <div data-chunk-name=\\"sample-component-3\\">
        <div>
          <h1>Sample Component</h1><pre>{
  &quot;test&quot;: {
    &quot;key&quot;: &quot;testValue&quot;
  },
  &quot;dr_pogodin_react_utils___split_components&quot;: {
    &quot;sample-component-1&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    },
    &quot;sample-component-2&quot;: {
      &quot;data&quot;: {},
      &quot;numRefs&quot;: 0,
      &quot;operationId&quot;: &quot;&quot;,
      &quot;timestamp&quot;: 1574985600000
    }
  }
}</pre>
        </div>
      </div>
    </div>
    <script id=\\"inj\\" type=\\"application/javascript\\" nonce=\\"abcdef-dummy-nonce\\">
      window.INJ = \\"MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEMKtsRsLh8R6HHpitZ/eghsfwaFoaus6qLoke9MlgqF39wUyiJOuxKmqjECLiFel3AfVxBlTkA2sj6FNYLq9mBqrVlCGhUyhXTWTPFlSbgmX7mf9UJCV6o+6N49TRM8T/Zt5SRase/W0YFEHYmiKSLZgUQPXsUoxsSTABdZa9BdRV/oUK68Qi2md0HR0J+b+HTZLAIsrRxHKMjBsUfmWUQjhOXxz8uZkjMyqgSptcNXEftmR5kazR/qzkkQcObSKPuDAKU03KPvQYU8IFE4CiNOX2Iw+T3MoeOLcuMcbBkvEUpXOo8tg7/9XQIY2QxEY+VOgOOywhCiFCspUah8KD+PeKXkuJdPx731E90C7TJz6AAvBfqtI2xCmRQrm5eP/Hk+E+WziLt4iuX0A+KQf2GhT7h2B6WWz+OiLd4i7p2hLaS8dJOsWjmKmAPz8mqTlA=\\"
    </script>
    <script src=\\"/test/public/path/polyfills-1511941200000.js\\" type=\\"application/javascript\\"></script>
    <script src=\\"/test/public/path/main-1511941200000.js\\" type=\\"application/javascript\\"></script>

  </body>

</html>"
`;
